### PR TITLE
Make possible to create KafkaProducerRecord with headers

### PR DESCRIPTION
**Pull request by bnjdpn**
From this question : https://github.com/vert-x3/vertx-kafka-client/issues/99

### CODE REVIEW

**Comment by bnjdpn** [comment]
@ppatierno Can you please take a look :) ?

File: src/main/java/io/vertx/kafka/client/producer/KafkaHeader.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.vertx.kafka.client.producer;
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/client/producer/KafkaHeader.java]: Add the license header, please.

File: src/test/java/io/vertx/kafka/client/tests/KafkaHeaderImplTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package io.vertx.kafka.client.tests;
```
Comment (ppatierno) [inline src/test/java/io/vertx/kafka/client/tests/KafkaHeaderImplTest.java]: Add the license header, please.

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package io.vertx.kafka.client.producer.impl;
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java]: Add the license header, please.

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  public String key() {
+    return key;
+  }
+
+  @Override
+  public byte[] value() {
+    return value;
+  }
+
+  public static List<KafkaHeader> fromHeaders(Headers headers) {
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java]: Add Java doc please.

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    if (headers == null || headers.toArray().length == 0) {
+      return Collections.emptyList();
+    }
+
+    return Stream.of(headers.toArray()).map(header -> new KafkaHeaderImpl(header.key(), header.value())).collect(Collectors.toList());
+  }
+
+  public static List<Header> toHeaderList(List<KafkaHeader> kafkaHeaders) {
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java]: Add Java doc please.

File: src/main/java/io/vertx/kafka/client/producer/KafkaHeader.java (hunk 1 above)
Comment (bnjdpn) [reply src/main/java/io/vertx/kafka/client/producer/KafkaHeader.java]: Done

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java (hunk 3 above)
Comment (bnjdpn) [reply src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java]: Done

File: src/test/java/io/vertx/kafka/client/tests/KafkaHeaderImplTest.java (hunk 2 above)
Comment (bnjdpn) [reply src/test/java/io/vertx/kafka/client/tests/KafkaHeaderImplTest.java]: Done

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java (hunk 4 above)
Comment (bnjdpn) [reply src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java]: Done

File: src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java (hunk 5 above)
Comment (bnjdpn) [reply src/main/java/io/vertx/kafka/client/producer/impl/KafkaHeaderImpl.java]: Done

**Comment by bnjdpn** [comment]
@ppatierno Thanks for the review ! I did what you wanted, same for the Javadoc ;)

**Comment by bnjdpn** [comment]
@ppatierno Ok. I just added new tests. What do you think of it ?

**Comment by bnjdpn** [comment]
@ppatierno Thanks, but I just saw that the build is KO after the merge... Do you know I can fix it ?

For this error :

> [ERROR] diagnostic: /scratch/jenkins/workspace/vert.x3-kafka-client/src/main/java/io/vertx/kafka/client/producer/KafkaProducerRecord.java:30: error: Could not generate model for io.vertx.kafka.client.producer.KafkaProducerRecord: Failed to generate because it would be impossible in target language JS at runtime to resolve which of the following overloaded methods to call in the Java API: create(java.lang.String topic, K key, V value) and create(java.lang.String topic, V value, java.util.List<io.vertx.kafka.client.producer.KafkaHeader> kafkaHeaders)

Maybe I should rename the method `create(java.lang.String topic, V value, java.util.List<io.vertx.kafka.client.producer.KafkaHeader> kafkaHeaders)` to `createWithHeaders(...)`

But for the second one : 

> [ERROR] diagnostic: /scratch/jenkins/workspace/vert.x3-kafka-client/src/main/java/io/vertx/kafka/client/producer/KafkaHeader.java:29: error: Could not generate model for io.vertx.kafka.client.producer.KafkaHeader#value(): Illegal type byte[] of kind ARRAY

I have a no idea. Maybe I should change the type to `String`.

What do you think ?

**Pull request merged**